SnipWiz/snipwiz.cpp

```diff
@@ -113,18 +113,23 @@ SnipWiz::SnipWiz(IManager* manager)
     m_shortName = plugName;
     m_topWin = m_mgr->GetTheApp();
 
-    // get plugin path
-    m_pluginPath = m_mgr->GetStartupDirectory();
-    m_pluginPath += wxFILE_SEP_PATH;
-    m_pluginPath += wxT("templates");
-    m_pluginPath += wxFILE_SEP_PATH;
-    if(!wxFileName::DirExists(m_pluginPath)) {
-        wxFileName::Mkdir(m_pluginPath);
-    }
+    // get config path
+    m_configPath = clStandardPaths::Get().GetUserDataDir();
+    m_configPath += wxFILE_SEP_PATH;
+    m_configPath += wxT("config");
+    m_configPath += wxFILE_SEP_PATH;
 
     m_StringDb.SetCompress(true);
 
-    m_StringDb.Load(m_pluginPath + defaultTmplFile);
+    if(!m_StringDb.Load(m_configPath + defaultTmplFile)) {
+        // For compatibility with CodeLite < 15.0.4:
+        // we don't use this directory for storing templates anymore.
+        wxString pluginPath = m_mgr->GetStartupDirectory();
+        pluginPath += wxFILE_SEP_PATH;
+        pluginPath += wxT("templates");
+        pluginPath += wxFILE_SEP_PATH;
+        m_StringDb.Load(pluginPath + defaultTmplFile);

```

Comment:
I would add code here to save it in the new location. so on next startup, it won't access this folder again:

```c++
// now save it in its new location
m_StringDb.Save(m_configPath + defaultTmplFile);
```

Reply:
Done, I've changed it to always save the new one in destructor (when CodeLite quits).